unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -414,7 +413,7 @@ public int indexOf(UTF8String v, int start) {
    */
   public UTF8String rpad(int len, UTF8String pad) {
     int spaces = len - this.numChars(); // number of char need to pad
-    if (spaces <= 0) {
+    if (spaces <= 0 || pad.numChars() == 0) {
       // no padding at all, return the substring of the current string
       return substring(0, len);
```

Comment:
@chenghao-intel and I think it's more proper to return the original string when `pad` is an empty string. Hive didn't handle this situation and just hangs forever.